python/lsst/ts/tcpip/one_client_read_loop_server.py

```diff
@@ -0,0 +1,143 @@
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import abc
+import asyncio
+import collections.abc
+import ctypes
+import logging
+import typing
+
+from .base_client_or_server import BaseClientOrServer, ConnectCallbackType
+from .constants import TERMINATOR
+from .one_client_server import OneClientServer
+
+__all__ = ["OneClientReadLoopServer"]
+
+
+class OneClientReadLoopServer(OneClientServer):
+    """A OneClientServer that reads and processes data in a loop.
+
+    Parameters
+    ----------
+    host : `str` or `None`
+        IP address for this server; typically `LOCALHOST_IPV4` for IPV4
+        or `LOCALHOST_IPV6` for IPV6. If `None` then bind to all network
+        interfaces (e.g. listen on an IPv4 socket and an IPv6 socket).
+        None can cause trouble with port=0;
+        see ``port`` in the Attributes section for more information.
+    port : `int`
+        IP port for this server. If 0 then randomly pick an available port
+        (or ports, if listening on multiple sockets).
+        0 is strongly recommended for unit tests.
+    log : `logging.Logger`
+        Logger.
+    connect_callback : callable or `None`, optional
+        Asynchronous or (deprecated) synchronous function to call when
+        when a client connects or disconnects.
+        If the other end (client) closes the connection, it may take
+        ``monitor_connection_interval`` seconds or longer to notice.
+        The function receives one argument: this `OneClientServer`.
+    name : `str`, optional
+        Name used for log messages, e.g. "Commands" or "Telemetry".
+    **kwargs : `dict` [`str`, `typing.Any`]
+        Additional keyword arguments for `asyncio.start_server`,
+        beyond host and port.
+
+    Attributes
+    ----------
+    read_loop_task : `asyncio.Future`
+        A task that reads messages from the reader in a loop.
```

Comment:
Consider `data` for consistency -- you use the variable name `data` instead of `message` in most places.

In fact I suggest a global search for `message`

Reply:
You still have "messages" instead of "data" or "items of data" in this doc?